vector/src/main/java/im/vector/app/features/home/room/list/home/release/ReleaseNotesFragment.kt

```diff
@@ -0,0 +1,126 @@
+                .also { it.attach() }
+
+        views.releaseNotesCarousel.registerOnPageChangeCallback(object : ViewPager2.OnPageChangeCallback() {
+            override fun onPageSelected(position: Int) {
+                viewModel.handle(ReleaseNotesAction.PageSelected(position))
+                updateButtonText(position)
+            }
+        })
+
+        carouselController.setData(createCarouselData())
+
+        views.releaseNotesBtnClose.onClick { close() }
+        views.releaseNotesButtonNext.onClick {
+            val isLastItemSelected = with(views.releaseNotesCarouselIndicator) {
+                selectedTabPosition == tabCount - 1
+            }
+            viewModel.handle(ReleaseNotesAction.NextPressed(isLastItemSelected))
+        }
+
+        viewModel.observeViewEvents {
+            when (it) {
+                is ReleaseNotesViewEvents.SelectPage -> selectPage(it.index)
+                ReleaseNotesViewEvents.Close -> close()
+            }
+        }
+    }
+
+    private fun createCarouselData(): ReleaseCarouselData {
+        return ReleaseCarouselData(
+                listOf(
+                        ReleaseCarouselData.Item(
+                                R.string.onboarding_new_app_layout_welcome_title,
+                                R.string.onboarding_new_app_layout_welcome_message,
+                                R.drawable.app_layout_onboarding_welcome
+                        ),
+                        ReleaseCarouselData.Item(
+                                R.string.onboarding_new_app_layout_spaces_title,
+                                R.string.onboarding_new_app_layout_spaces_message,
+                                R.drawable.app_layout_onboarding_spaces
+                        ),
+                        ReleaseCarouselData.Item(
+                                R.string.onboarding_new_app_layout_feedback_title,
+                                R.string.onboarding_new_app_layout_feedback_message,
+                                R.drawable.app_layout_onboarding_feedback
+                        ),
+                )
+        )
+    }
+
+    private fun close() {
+        requireActivity().onBackPressed()
+    }
+
+    private fun selectPage(index: Int) {
+        views.releaseNotesCarouselIndicator.selectTab(views.releaseNotesCarouselIndicator.getTabAt(index))
+        updateButtonText(index)
+    }
+
+    private fun updateButtonText(selectedIndex: Int) {
+        val isLastItem = selectedIndex == views.releaseNotesCarouselIndicator.tabCount - 1
+        if (isLastItem) {
+            views.releaseNotesButtonNext.setText(R.string.onboarding_new_app_layout_button_try)
+        } else {
+            views.releaseNotesButtonNext.setText(R.string.action_next)
+        }
+    }
+
+    override fun onDestroyView() {
+        tabLayoutMediator?.detach()
+        tabLayoutMediator = null
+
+        views.releaseNotesCarousel.adapter = null
```

Comment:
You should also invoke `views.releaseNotesCarousel.unregisterOnPageChangeCallback`. Not sure they will be a memory leak, but let's not take the risk.